# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsDragServiceProxy.cpp
# Commit: 0df2f2832755
# Full Hash: 0df2f283275530fbfe5cf9c95986b073c0d6a31e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 1d21d911b3e7 -r 0df2f2832755 widget/nsDragServiceProxy.cpp
--- a/widget/nsDragServiceProxy.cpp	Mon Aug 22 15:35:28 2022 +0000
+++ b/widget/nsDragServiceProxy.cpp	Mon Aug 22 15:38:14 2022 +0000
@@ -80,8 +80,8 @@
         }
 
         mozilla::Unused << child->SendInvokeDragSession(
-            dataTransfers, aActionType, Some(std::move(surfaceData)), stride,
-            dataSurface->GetFormat(), dragRect, principal, csp, csArgs,
+            std::move(dataTransfers), aActionType, Some(std::move(surfaceData)),
+            stride, dataSurface->GetFormat(), dragRect, principal, csp, csArgs,
             mSourceWindowContext);
         StartDragSession();
         return NS_OK;
@@ -90,8 +90,9 @@
   }
 
   mozilla::Unused << child->SendInvokeDragSession(
-      dataTransfers, aActionType, Nothing(), 0, static_cast<SurfaceFormat>(0),
-      dragRect, principal, csp, csArgs, mSourceWindowContext);
+      std::move(dataTransfers), aActionType, Nothing(), 0,
+      static_cast<SurfaceFormat>(0), dragRect, principal, csp, csArgs,
+      mSourceWindowContext);
   StartDragSession();
   return NS_OK;
 }